### PR TITLE
TfLiteCoreMlDelegateOptions doc updated #57565

### PR DESCRIPTION
**Pull request by Vaani-pathariya**
I tried to update the documentation

### CODE REVIEW

File: tensorflow/lite/delegates/coreml/README.md [hunk 1]
```diff
@@ -159,9 +159,30 @@ extension CoreMLDelegate {
...
+  // Create Core ML delegate only on devices with Apple Neural Engine.
+  // Returns nullptr otherwise.
+  TfLiteCoreMlDelegateDevicesWithNeuralEngine,
+  // Always create Core ML delegate
+  TfLiteCoreMlDelegateAllDevices
+} TfLiteCoreMlDelegateEnabledDevices;
 typedef struct {
- // We have dummy for now as we can't have empty struct in C.
- char dummy;
+// Only create delegate when Neural Engine is available on the device.
```
Comment (yishuangP) [inline tensorflow/lite/delegates/coreml/README.md:170]: 2 space indentation please.

**Review by yishuangP — APPROVED** [review]
Thanks for updating the documentation.

**Comment by gbaned** [comment]
Hi @Vaani-pathariya Can you please check @yishuangP's comments and keep us posted ? Thank you!

File: tensorflow/lite/delegates/coreml/README.md (hunk 1 above)
Comment (teijeong) [reply tensorflow/lite/delegates/coreml/README.md:170]: Please address the comment.

**Pull request closed without merging**